security/pkg/nodeagent/secretfetcher/secretfetcher.go

```diff
@@ -114,6 +121,54 @@ func (sf *SecretFetcher) Run(ch chan struct{}) {
 	cache.WaitForCacheSync(ch, sf.scrtController.HasSynced)
 }
 
+// Generate implements the XDS Generate method for SDS.
+// We also allow returning the secrets over the main XDS connection,
+// for efficiency.
+// This method only allows returning secret for authenticated requests.
+func (sf *SecretFetcher) Generate(proxy *istiomodel.Proxy, push *istiomodel.PushContext,
+	w *istiomodel.WatchedResource, updates istiomodel.XdsUpdates) istiomodel.Resources {
+	// 1. Get the namespace of the caller
+	// TODO: add an error to Generate (for non-empty)
+
+	// This depends on in-progress PR to VERIFY the namespace.
+
+	// 2. If istio-system - ok, can read secrets in istio-system
+	// This istiod can either be the 'master' in istio-system, or
+	// an replica of istiod running in a namespace with reduced
+	// RBAC to only namespace secret access. The second is equivalent
+	// with running the gateway with secret access, but secret permissions
+	// are granted to a separate istiod pod.
+	if proxy.Metadata.Namespace != sf.secretNamespace {
```

Comment:
I dont think this is helpful in all cases. Imagine the egress scenario where I have a destination rule with a tls cert to talk to some external service. I have my apps in all namespaces. and I have a ca cert bundle to verify the external service. Do I have to create this ca cert bundle in each and every namespace?

Reply:
I'm not even remotely trying to cover egress in this PR... Yes, we'll need a policy about which workload can use what secret for egress, or if they're not allowed force the request to use the egress gateway. 

But really the very first step is to get a SDS server in Istiod with the minimum set of features - replacing the direct access to secrets from the default gateway.